### PR TITLE
Feature flags setup

### PR DESCRIPTION
**Pull request by geoffreydhuyvetters**
### Added

- Simple hook to check if a feature flag is enabled and set it
- constants to use in code
- helper for the browser console
- custom 500 page

```js
setFeatureFlag(FeatureFlags.REACT_DASHBOARD, false)
```

### Updated

- Next to 10.1 and using Webpack 5 now

### CODE REVIEW

File: src/constants/FeatureFlags.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const FeatureFlags = {
+  REACT_DASHBOARD: 'react_dashboard',
+  REACT_CREATE: 'react_create',
+};
+
+export { FeatureFlags };
```
Comment (bertramakers) [inline src/constants/FeatureFlags.js:6]: Small nit pick: Do you think we'll use this constant somewhere without using the `useFeatureFlag` hook? If not I think we can put it together with the hook in the same file.

I think this `constants` folder was just added for constants that are used in multiple contexts, not related to one specific hook/component.

File: src/hooks/useFeatureFlag.js [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
 
   return [value, set];
 };
 
+if (typeof window !== 'undefined') {
+  window.FeatureFlags = FeatureFlags;
+  window.setFeatureFlag = (identifier, value) => {
+    cookies.set(createFullIdentifier(identifier), value);
+  };
+}
```
Comment (bertramakers) [inline src/hooks/useFeatureFlag.js:27]: No comment but just a question: I'm curious why we need these on the `window` object? To use them manually in the `console` or something?

File: src/constants/FeatureFlags.js (hunk 1 above)
Comment (simon-debruijn) [reply src/constants/FeatureFlags.js:6]: Yep I think this can also live in useFeatureFlags

**Review by bertramakers — APPROVED** [review]

File: src/hooks/useFeatureFlag.js (hunk 2 above)
Comment (bertramakers) [reply src/hooks/useFeatureFlag.js]: Nevermind I saw your explanation on Slack now 👍

**Review by simon-debruijn — APPROVED** [review]
Would move `FeatureFlags` constant to `usefeatureFlag.js`.

The rest looks good 👍🏻

File: src/constants/FeatureFlags.js (hunk 1 above)
Comment (geoffreydhuyvetters) [reply src/constants/FeatureFlags.js:6]: Jep, good idea

**Review by jdrieghe — APPROVED** [review]
image

File: src/hooks/useFeatureFlag.js (hunk 2 above)
Comment (geoffreydhuyvetters) [reply src/hooks/useFeatureFlag.js]: idd, exposing a function on window makes them 'global' to use in the console

**Comment by bertramakers** [comment]
The Chromatic build seems to be failing because it cannot read the apiKey config (which is undefined at build time). I think this will also be a problem when we build it on Jenkins, or at the latest when we deploy it and run it with runtime config on the servers. Because Next seems to be reading the config at build-time again. Is this something that changed in the Next update maybe?

**Comment by simon-debruijn** [comment]
> The Chromatic build seems to be failing because the Next app cannot read the apiKey config (which is undefined at build time). I think this will also be a problem when we build the app on Jenkins, or at the latest when we deploy it and run it with runtime config on the servers. Because Next seems to be reading the config at build-time again. Is this something that changed in the Next update maybe?

Yep I think it has been changed.

We can just call `process.env.NEXT_PUBLIC_MY_VAR` it seems. In stead of having to use `next/config`
https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

**Comment by geoffreydhuyvetters** [comment]
that's not new since 10.1 @simon-debruijn, we tried it like that before (on 10.0.x) but for some reason that didn't work in our setup.

Not sure what's happening now to be fair 🤔 Let me have a look.

**Comment by simon-debruijn** [comment]
> that's not new since 10.1 @simon-debruijn, we tried it like that before (on 10.0.x) but for some reason that didn't work in our setup. I think @bertramakers can provide a bit of context there.
> 
> Not sure what's happening now to be fair 🤔 Let me have a look.

Hmm ok strange

**Comment by bertramakers** [comment]
I think `process.env.NEXT_PUBLIC_MY_VAR` is for exposing environment variables to the browser, but their values will still be "hardcoded" into the build files. Both `process.env.NEXT_PUBLIC_...` and `publicRuntimeConfig` are public, but we also need it to be runtime config specifically.

**Comment by geoffreydhuyvetters** [comment]
Exactly, that was it. It's causing issues on the 500 page, which is new to 10.1. At least I have a direction to look into.

**Comment by bertramakers** [comment]
> Exactly, that was it. It's causing issues on the 500 page, which is new to 10.1. At least I have a direction to look into.

I seem to remember we also had that issue with the 404 page. Next is trying to render them as static pages. So we forced the 404 page to be rendered server-side instead (vs at build time) I think.

**Pull request merged**